app/routes.py

```diff
@@ -1,2 +1,209 @@
+def get_video(active_id):
+    video = Video.query.get_or_404(active_id)
+    return make_response(video.to_dict(), 200)
+
+@videos_bp.route("/<active_id>", methods=["PUT"])
+def put_video(active_id):
+    video = Video.query.get_or_404(active_id)
+    request_body = request.get_json()
+
+    try:
+        video.title=request_body["title"],
+        video.release_date=request_body["release_date"],
+        video.total_inventory=request_body["total_inventory"]
+    except KeyError:
+        return make_response({"details" : "Invalid data"}, 400)
+
+    db.session.commit()
+    return make_response(video.to_dict(), 200)
+
+@videos_bp.route("/<active_id>", methods=["DELETE"])
+def delete_video(active_id):
+    video = Video.query.get_or_404(active_id)
+    db.session.delete(video)
+    db.session.commit()
+    return ({"id" : int(active_id)}, 200)
+
+@videos_bp.route("/<active_id>/rentals", methods=["GET"])
+def get_rentals_by_video(active_id):
+    video = Video.query.get_or_404(active_id)
+
+    rental_list = []
+    for rental in video.active_rentals:
+        customer = Customer.query.get_or_404(rental.customer_id)
+        rental_list.append({"name" : customer.name,
+            "phone" : customer.phone,
+            "postal_code" : customer.postal_code,
+            "due_date" : rental.due_date
+            })
+
+    return jsonify(rental_list)
+
+# ===== Rentals =====================================================
+rentals_bp = Blueprint("rental", __name__, url_prefix="/rentals")
+
+@rentals_bp.route("/check-out", methods=["POST"])
+def check_out_rental():
+    request_body = request.get_json()
+
+    try:
+        if type(request_body["customer_id"]) is not int or type(request_body["video_id"]) is not int:
+            return make_response({"details" : "Invalid data"}, 400)
+    except KeyError:
+        return make_response({"details" : "Invalid data"}, 400)
+
+    customer = Customer.query.get_or_404(request_body["customer_id"])
+    video = Video.query.get_or_404(request_body["video_id"])
+
+    if (video.total_inventory - len(video.active_rentals)) < 1:
+        return make_response({"details" : "Not available"}, 400)
+
+    new_rental = Rental(
+            customer_id=customer.id,
+            video_id=video.id,
+            due_date=(datetime.now() + timedelta(days=7))
+        )
+
+    db.session.add(new_rental)
+    db.session.commit()
+
+    ret_body = new_rental.to_dict()
+    ret_body["videos_checked_out_count"] = len(customer.active_rentals)
+    ret_body["available_inventory"] = video.total_inventory - len(video.active_rentals)
+
+    return make_response(ret_body, 200)
+
+@rentals_bp.route("/check-in", methods=["POST"])
+def check_in_rental():
+    request_body = request.get_json()
+
+    try:
+        rental = Rental.query.filter_by(
+            customer_id=request_body["customer_id"], 
+            video_id=request_body["video_id"]
+            ).first_or_404(description="No active rental for this customer and video")
```

Comment:
This works, but it fails the smoke test because the test was expecting a status of 400 & a JSON response.